apps/music/test/marionette/metadata_test.js

```diff
@@ -0,0 +1,73 @@
+/* global require, marionette, setup, suite, test, __dirname */
+'use strict';
+
+var assert = require('assert');
+var Music = require('./lib/music.js');
+var FakeRingtones = require('./lib/fakeringtones.js');
+var FakeControls = require('./lib/fakecontrols.js');
+var ListviewHelper = require('./lib/listviewhelper.js');
+
+marionette('Music player playlist', function() {
+  var apps = {};
+  apps[FakeRingtones.DEFAULT_ORIGIN] = __dirname + '/fakeringtones';
+  apps[FakeControls.DEFAULT_ORIGIN] = __dirname + '/fakecontrols';
+
+  var client = marionette.client({
+    prefs: {
+      'device.storage.enabled': true,
+      'device.storage.testing': true,
+      'device.storage.prompt.testing': true
+    },
+
+    settings: {
+      'lockscreen.enabled': false,
+      'ftu.manifestURL': null
+    },
+
+    apps: apps
+  });
+
+  var music;
+
+  setup(function() {
+    music = new Music(client);
+
+  });
+
+  suite('Album art display', function () {
+    setup(function() {
+      client.fileManager.removeAllFiles();
+      client.fileManager.add([
+        {
+          type: 'music',
+          filePath: 'apps/music/test-data/playlists/a.ogg'
+        }
+      ]);
+    });
+
+
+    test('Check album art is displayed', function() {
+
+      music.launch();
+      music.waitForFirstTile();
+
+      music.switchToArtistsView();
+
+      var bg = ListviewHelper.backgroundImageForListItem(music.listItems, 0);
+      // Test that the album cover is a blob. This mean it comes
+      // from the metadata.
+      assert.ok(bg.indexOf('url("blob:') != -1);
+
+      music.switchToAlbumsView();
+
+      bg = ListviewHelper.backgroundImageForListItem(music.listItems, 0);
+      assert.ok(bg.indexOf('url("blob:') != -1);
+
+      music.switchToSongsView();
+
+      bg = ListviewHelper.backgroundImageForListItem(music.listItems, 0);
+      assert.ok(bg.indexOf('url("blob:') != -1);
+    });
+
```

Comment:
This would be a more convincing test if you had two songs with different album art, and you verified that each song had the same blob url in each of the views and that the two songs always had different blob urls.  Or at least have one song with album art and one without and verify that they display different urls.

Of course, the ideal thing would be to verify that the album art was actually correct. Like make on song have an image that is a solid black square and the other have a solid white square and then somehow check that the right colors are displayed. I don't know how you could actually do that, though, so I don't want to require it here.